presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveBasicTableStatistics.java

```diff
@@ -0,0 +1,473 @@
+            assertThatStatisticsAreZero(statisticsAfterSecondDelete);
+        }
+        finally {
+            onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+        }
+    }
+
+    private static void insertNationData(QueryExecutor executor, String tableName)
+    {
+        executor.executeQuery(format("" +
+                "INSERT INTO %s (n_nationkey, n_regionkey, n_name, n_comment) " +
+                "SELECT n_nationkey, n_regionkey, n_name, n_comment FROM nation", tableName));
+    }
+
+    private static void assertThatStatisticsArePresent(BasicStatistics statistics)
+    {
+        assertThat(statistics.getNumRows()).isPresent();
+        assertThat(statistics.getNumFiles()).isPresent();
+        assertThat(statistics.getRawDataSize()).isPresent();
+        assertThat(statistics.getTotalSize()).isPresent();
+    }
+
+    private static void assertThatStatisticsAreNonZero(BasicStatistics statistics)
+    {
+        assertThat(statistics.getNumRows().getAsLong()).isGreaterThan(0);
+        assertThat(statistics.getNumFiles().getAsLong()).isGreaterThan(0);
+        assertThat(statistics.getRawDataSize().getAsLong()).isGreaterThan(0);
+        assertThat(statistics.getTotalSize().getAsLong()).isGreaterThan(0);
+    }
+
+    private static void assertThatStatisticsAreZero(BasicStatistics statistics)
+    {
+        assertThat(statistics.getNumRows().getAsLong()).isEqualTo(0);
+        assertThat(statistics.getNumFiles().getAsLong()).isEqualTo(0);
+        assertThat(statistics.getRawDataSize().getAsLong()).isEqualTo(0);
+        assertThat(statistics.getTotalSize().getAsLong()).isEqualTo(0);
+    }
+
+    private static void assertThatStatisticsValuesHaveIncreased(BasicStatistics first, BasicStatistics second)
+    {
+        assertThat(second.getNumRows().getAsLong()).isGreaterThan(first.getNumRows().getAsLong());
+        assertThat(second.getNumFiles().getAsLong()).isGreaterThan(first.getNumFiles().getAsLong());
+        assertThat(second.getTotalSize().getAsLong()).isGreaterThan(first.getTotalSize().getAsLong());
+        assertThat(second.getRawDataSize().getAsLong()).isGreaterThan(first.getRawDataSize().getAsLong());
+    }
+
+    private static void assertThatStatisticsValuesHaveDecreased(BasicStatistics first, BasicStatistics second)
+    {
+        assertThat(second.getNumRows().getAsLong()).isLessThan(first.getNumRows().getAsLong());
+        assertThat(second.getNumFiles().getAsLong()).isLessThan(first.getNumFiles().getAsLong());
+        assertThat(second.getTotalSize().getAsLong()).isLessThan(first.getTotalSize().getAsLong());
+        assertThat(second.getRawDataSize().getAsLong()).isLessThan(first.getRawDataSize().getAsLong());
+    }
+
+    private static BasicStatistics getBasicStatisticsForTable(QueryExecutor executor, String table)
+    {
+        QueryResult result = executor.executeQuery(format("DESCRIBE FORMATTED %s", table));
```

Comment:
`s/executor/onHive()` this query is is not going to run with any there query exeutor

Reply:
Since there's a lot of interaction between Hive and Presto i though it will be easier to read it when you can see where the query is executed at every single step. I don't have a strong opinion though.